Index: dahdi-tools-2.6.0/xpp/Makefile
===================================================================
--- dahdi-tools-2.6.0.orig/xpp/Makefile	2011-07-11 00:25:18.000000000 +0800
+++ dahdi-tools-2.6.0/xpp/Makefile	2012-07-21 23:30:58.244897984 +0800
@@ -28,7 +28,7 @@
 DATADIR		= $(datadir)/dahdi
 MANDIR		= $(mandir)/man8
 HOTPLUG_USB_DIR	= $(sysconfdir)/hotplug/usb
-PERLLIBDIR	:= $(shell eval `perl -V:sitelib`; echo "$$sitelib")
+PERLLIBDIR	= $(DAHDI_PERLLIBDIR)
 PERL_DIRS	:= $(shell cd perl_modules; find * -name '[A-Z]*' -type d| xargs)
 PERL_MODS_PAT	:= *.pm $(PERL_DIRS:%=%/*.pm)
 PERL_MODS	:= $(shell cd perl_modules; echo $(PERL_MODS_PAT))
