diff --git a/configure.ac b/configure.ac
index 1911187..c395ee2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -185,8 +185,8 @@ AM_CONDITIONAL(ENABLE_GSM_MS, test "x$with_gsm_ms" == "xyes" )
 AM_CONDITIONAL(ENABLE_GSM, test "x$with_gsm_bs" == "xyes" -o "x$with_gsm_ms" == "xyes")
 
 # check for opencore-amrnb for AMR and EFR decoding
-found_opencore_amrnb=yes
-PKG_CHECK_MODULES(OPENCORE_AMRNB, opencore-amrnb >= 0.1.0, , found_opencore_amrnb=no)
+found_opencore_amrnb=no
+#PKG_CHECK_MODULES(OPENCORE_AMRNB, opencore-amrnb >= 0.1.0, , found_opencore_amrnb=no)
 AM_CONDITIONAL(ENABLE_GSMAMR, test "$found_opencore_amrnb" = "yes")
 if test "$found_opencore_amrnb" = yes; then
 	AC_DEFINE(HAVE_OPENCORE_AMRNB, 1, [Define to 1 if OpenCore AMR-NB library is available])
