diff --git a/gtp/queue.h b/gtp/queue.h
index 556b6ef..d59a1a3 100644
--- a/gtp/queue.h
+++ b/gtp/queue.h
@@ -19,7 +19,7 @@
 
 #define QUEUE_DEBUG 0		/* Print debug information */
 
-#define QUEUE_SIZE 1024		/* Size of retransmission queue */
+#define QUEUE_SIZE 32		/* Size of retransmission queue */
 #define QUEUE_HASH_SIZE 65536	/* Size of hash table (2^16) */
 
 struct qmsg_t {			/* Holder for queued packets */
